src/util.cpp

```diff
@@ -889,6 +889,99 @@ QSize CMinimumStackedLayout::sizeHint() const
 * Other Classes                                                                *
 \******************************************************************************/
 // Network utility functions ---------------------------------------------------
+bool NetworkUtil::ParseNetworkAddressString ( QString strAddress, QHostAddress& InetAddr, bool bEnableIPv6 )
+{
+    // try to get host by name, assuming
+    // that the string contains a valid host name string or IP address
+    const QHostInfo HostInfo = QHostInfo::fromName ( strAddress );
+
+    if ( HostInfo.error() != QHostInfo::NoError )
+    {
+        // qInfo() << qUtf8Printable ( QString ( "Invalid hostname" ) );
+        return false; // invalid address
+    }
+
+    foreach ( const QHostAddress HostAddr, HostInfo.addresses() )
+    {
+        // qInfo() << qUtf8Printable ( QString ( "Resolved network address to %1 for proto %2" ) .arg ( HostAddr.toString() ) .arg (
+        // HostAddr.protocol() ) );
+        if ( HostAddr.protocol() == QAbstractSocket::IPv4Protocol || ( bEnableIPv6 && HostAddr.protocol() == QAbstractSocket::IPv6Protocol ) )
+        {
+            InetAddr = HostAddr;
+            return true;
+        }
+    }
+    return false;
+}
+
+#ifndef CLIENT_NO_SRV_CONNECT
+bool NetworkUtil::ParseNetworkAddressSrv ( QString strAddress, CHostAddress& HostAddress, bool bEnableIPv6 )
+{
+    // init requested host address with invalid address first
+    HostAddress = CHostAddress();
+
+    QRegularExpression plainHostRegex ( "^([^\\[:0-9.][^:]*)$" );
+    if ( plainHostRegex.match ( strAddress ).capturedStart() != 0 )
+    {
+        // not a plain hostname? then don't attempt SRV lookup and fail
+        // immediately.
+        return false;
+    }
+
+    QDnsLookup* dns = new QDnsLookup();
+    dns->setType ( QDnsLookup::SRV );
+    dns->setName ( QString ( "_jamulus._udp.%1" ).arg ( strAddress ) );
+    dns->lookup();
+    // QDnsLookup::lookup() works asynchronously. Therefore, wait for
+    // it to complete here by resuming the main loop here.
+    // This is not nice and blocks the UI, but is similar to what
+    // the regular resolve function does as well.
+    QTime dieTime = QTime::currentTime().addMSecs ( DNS_SRV_RESOLVE_TIMEOUT_MS );
+    while ( QTime::currentTime() < dieTime && !dns->isFinished() )
```

Comment:
Not a fan of busy waiting (even if it's limited). Isn't there any blocking function or something like a signal?

Reply:
You're asking for a Qt equivalent of AA/AAAA lookup for SRV names?

Reply:
> Isn't there any blocking function

The existing code path uses [QHostInfo::fromName](https://doc.qt.io/qt-6/qhostinfo.html#fromName), which is a blocking function. I haven't found any equivalent for SRV lookups.

> or something like a signal?

The `QDnsLookup`, which I've used here, is supposed to be used via signals. However, this requires -- as far as I can see -- an inversion of the control flow as the existing logic relies on blocking. Even when that refactoring is done, new questions come up: How should the UI behave? Currently it blocks. If it continues to stay responsive during lookup, we would need some other kind of indicator, so that's kind of a rabbit hole. Not saying that it shouldn't be done, but rather that it's a larger refactoring which I didn't want to do as part of this PR.

In short: That's the best and least intrusive way I've found. If someone finds a better way, I'm certainly open to it. :)